### ISSUE TITLE
AP_DDS: Provide current Goal

### ISSUE THREAD

**Issue by tizianofiorenzani** [ArduPilot/ardupilot#28299]
## Provide current goal location to the DDS Interface

Providing the current goal would help to verify whether the vehicle is actually going where it has been guided to.

**Proposed Solution**
- Create a new topic `/ap/goal` of type GeoPoseStamped, and use it to send the goal location. 
- The goal location in the DDS client should be recovered with `AP::vehicle()->get_target_location(Location &target_loc);`

**Altitude frame**
- Goal could be either be converted to MSL or we could provide the altitude frame in the `frame_id`.

**Platform**
[  ] All
[  ] AntennaTracker
[ x] Copter
[ x] Plane
[ x] Rover
[ x] Submarine

**Comment by Ryanf55** [comment]
In what situation does a companion computer not know what where the autopilot is headed? Won't all commands come from the companion computer over reliable transport?

**Comment by tizianofiorenzani** [comment]
Not really, as the goal is sent by topic, so you trust that the vehicle obeys, which I find it too flimsy.

**Comment by tizianofiorenzani** [comment]
Created a PR: https://github.com/ArduPilot/ardupilot/pull/28372

**Comment by peterbarker** [comment]
Well, we merged the PR, so closing this one :-)
